core/src/main/java/tech/pegasys/eth2signer/core/Runner.java

```diff
@@ -56,27 +59,43 @@ public Runner(final Config config) {
   public void run() {
 
     final Vertx vertx = Vertx.vertx();
+    final MetricsEndpoint metricsEndpoint =
+        new MetricsEndpoint(
+            config.isMetricsEnabled(),
+            config.getMetricsPort(),
+            config.getMetricsNetworkInterface(),
+            config.getMetricCategories(),
+            vertx);
     try {
-      final Handler<HttpServerRequest> requestHandler = createRouter(vertx);
+      final Handler<HttpServerRequest> requestHandler =
+          createRouter(vertx, metricsEndpoint.getMetricsSystem());
       final HttpServer httpServer = createServerAndWait(vertx, requestHandler);
       LOG.info("Server is up, and listening on {}", httpServer.actualPort());
 
+      metricsEndpoint.start();
       persistPortInformation(httpServer.actualPort());
     } catch (final ExecutionException | InterruptedException e) {
       vertx.close();
       throw new RuntimeException("Failed to create Http Server", e.getCause());
     } catch (final Throwable t) {
+      metricsEndpoint.stop();
```

Comment:
nit:metricsEndpoing _may_ not have started - is that a problem?

Reply:
No shouldn't be a problem. The Besu MetricsHttpService does a null check on the internal Vertx httpServer before attempting to do anything.